record/record.go

```diff
@@ -169,9 +179,10 @@ func (r *Reader) nextChunk(wantFirst bool) error {
 				}
 			}
 			r.last = chunkType == fullChunkType || chunkType == lastChunkType
+			r.recovering = false
 			return nil
 		}
-		if r.n < blockSize && r.started {
+		if r.n < blockSize && r.started && !r.recovering {
```

Comment:
I actually don't think we need a test on !r.recovering here at all. I've been thinking about it. If r.n is less than blockSize, we know there are no more blocks at all.  Thus we are logically at some sort EOF whether r.recovering is true or not.

That said, if r.j != r.n, there is trailing data in the last block of the file which is junk.  Thus io.ErrUnexpectedEOF is valid.  If, on the other hand, they are equal, we'll hit the EOF case.

Put another way, once we meet the condition r.n < blockSize && r.started  there is no possible way to recover, and if there is no more data, the conditional in this block of code is sufficient, imho.

WDYT?